echopype/visualize/api.py

```diff
@@ -0,0 +1,123 @@
+        'long_name': 'Range',
+        'units': 'm',
+    }
+
+    if isinstance(frequency, list) and len(frequency) == 1:
+        frequency = frequency[0]
+
+    if isinstance(data, EchoData):
+        if data.sonar_model.lower() == 'ad2cp':
+            raise ValueError(
+                "Visualization for AD2CP sonar model is currently unsupported."
+            )
+        yaxis = 'range_bin'
+        variable = 'backscatter_r'
+        ds = data.beam
+        if get_range:
+            yaxis = 'range'
+
+            if data.sonar_model.lower() == 'azfp':
+                if 'azfp_cal_type' not in range_kwargs:
+                    range_kwargs['azfp_cal_type'] = 'Sv'
+                if 'env_params' not in range_kwargs:
+                    raise ValueError(
+                        "Please provide env_params in range_kwargs!"
+                    )
+            elif data.sonar_model.lower() == 'ek60':
+                if 'waveform_mode' not in range_kwargs:
+                    range_kwargs['waveform_mode'] = 'CW'
+
+                if 'encode_mode' not in range_kwargs:
+                    range_kwargs['encode_mode'] = 'power'
+            elif data.sonar_model.lower() == 'ek80':
+                if 'waveform_mode' not in range_kwargs:
+                    raise ValueError(
+                        "Please provide waveform_mode in range_kwargs for EK80 sonar model."
+                    )
+
+            range_in_meter = data.compute_range(
+                env_params=range_kwargs.get('env_params', {}),
+                azfp_cal_type=range_kwargs.get('azfp_cal_type', None),
+                ek_waveform_mode=range_kwargs.get('waveform_mode', None),
+                ek_encode_mode=range_kwargs.get('encode_mode', 'complex'),
+            )
+            range_in_meter.attrs = range_attrs
+            if 'water_level' in data.platform:
+                # Adds water level to range if it exists
+                range_in_meter = range_in_meter + data.platform.water_level
+            ds = ds.assign_coords({'range': range_in_meter})
+            ds.range.attrs = range_attrs
+
+    elif isinstance(data, xr.Dataset):
+        variable = 'Sv'
+        ds = data
+        if get_range:
+            yaxis = 'range'
+        else:
+            if 'range' in data.dims:
+                # This indicates that data is MVBS.
+                yaxis = 'range'
+            else:
+                yaxis = 'range_bin'
+        # If depth is available in ds, use it.
+        ds = ds.set_coords('range')
+        if isinstance(water_level, xr.DataArray):
+            required_dims = ['frequency', 'ping_time']
+            if not all(
+                True if d in water_level.dims else False for d in required_dims
+            ):
+                raise ValueError(
+                    f"Water level must have dimensions: {', '.join(required_dims)}"
+                )
+            # Adds water level to range if it exists
+            ds['range'] = ds.range + water_level
```

Comment:
I think water level can be the following:
- a single number: this is often the case when all the transducers are mounted on a single plate on a ship or a mooring (in generally good weather so there's no significant change of depth).
- only have a single dimension `frequency`: in this case different transducers are mounted at different depths, but there was no change across `ping_time`
- only have a single dimension `ping_time`: in this case all transducers are mounted on the same plate, but there may be variations of depth for the entire platform (think about the case when you have an AUV with an echosounder on it)
- have both dimensions `frequency` and `ping_time` like what you have here

All of these should add to `ds.range` without problem since the dimension would be broadcast.